recipes/repl-builtin-modules/src/workflow.ts

```diff
@@ -0,0 +1,404 @@
+                        }
+                    }
+                }
+            }
+        } else {
+            // Handle default/namespace imports
+            const importClause = statement.find({ rule: { kind: "import_clause" } });
+            if (!importClause) continue;
+
+            // Use resolveBindingPath to determine how builtinModules should be accessed
+            const builtinModulesPath = resolveBindingPath(importClause, "$.builtinModules");
+            const builtinLibsPath = resolveBindingPath(importClause, "$._builtinLibs");
+
+            const expressions = rootNode.findAll({
+                rule: {
+                    any: [
+                        { pattern: builtinModulesPath },
+                        { pattern: builtinLibsPath }
+                    ]
+                }
+            });
+
+            if (expressions.length > 0) {
+                updateModuleSpecifier(statement, edits);
+
+                // Get the namespace identifier to maintain consistency
+                let importIdentifier = importClause.find({ rule: { kind: "identifier" } });
+                if (!importIdentifier) {
+                    const namespaceImport = importClause.find({
+                        rule: { kind: "namespace_import" }
+                    });
+
+                    if (namespaceImport) {
+                        importIdentifier = namespaceImport.find({ rule: { kind: "identifier" } });
+                    }
+                }
+
+                const varName = importIdentifier?.text() || "module";
+                for (const memberExpr of expressions) {
+                    edits.push(memberExpr.replace(`${varName}.builtinModules`));
+                }
+            }
+        }
+    }
+
+    // Step 3: Handle dynamic imports
+    const dynamicImportDeclarators = rootNode.findAll({
+        rule: {
+            kind: "variable_declarator",
+            has: {
+                field: "value",
+                kind: "await_expression",
+                has: {
+                    kind: "call_expression",
+                    all: [
+                        {
+                            has: {
+                                field: "function",
+                                kind: "import"
+                            }
+                        },
+                        {
+                            has: {
+                                field: "arguments",
+                                kind: "arguments",
+                                has: {
+                                    kind: "string",
+                                    has: {
+                                        kind: "string_fragment",
+                                        regex: "(node:)?repl$"
+                                    }
+                                }
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    });
```

Comment:
Woof 🥵 maybe extract this? when reading this code, I don't really care about this part: the const name sufficiently tells me what to expect (and if I really wanna see it, there's always <kbd>cmd</kbd> + click).